quickwit/quickwit-indexing/src/models/merging_statistics.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::sync::atomic::Ordering;
+
+use crate::actors::{PublisherCounters, UploaderCounters};
+
+/// A Struct to hold all merging statistical data.
+#[derive(Clone, Debug, Default)]
+pub struct MergingStatistics {
```

Comment:
We might be better served by Prometheus metrics for this. I'd get rid of it.

Reply:
As of today, these counters serve also the purpose of assertions in unit-tests. I would think it's ok to them keep around until replaced by the Prometheus metrics.